apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -115,3 +115,50 @@ function populate_wp_template_data() {
 	$fse->insert_default_data();
 }
 register_activation_hook( __FILE__, __NAMESPACE__ . '\populate_wp_template_data' );
+
+/**
+ * Add front-end CoBlocks gallery block scripts.
+ *
+ * This function performs the same enqueueing duties as `CoBlocks_Block_Assets::frontend_scripts`,
+ * but for our FSE header and footer content. `frontend_scripts` uses `has_block` to determine
+ * if gallery blocks are present, and `has_block` is not aware of content sections outside of
+ * post_content yet.
+ */
+function enqueue_coblocks_gallery_scripts() {
+	if ( ! function_exists( 'CoBlocks' ) ) {
+		return;
+	}
+
+	$template = new WP_Template();
+	$header   = $template->get_template_content( 'header' );
+	$footer   = $template->get_template_content( 'footer' );
+
+	// Define where the asset is loaded from.
+	$dir = CoBlocks()->asset_source( 'js' );
+
+	// Define where the vendor asset is loaded from.
+	$vendors_dir = CoBlocks()->asset_source( 'js', 'vendors' );
+
+	// Masonry block.
+	if ( has_block( 'coblocks/gallery-masonry', $header . $footer ) ) {
+		wp_enqueue_script(
+			'coblocks-masonry',
+			$dir . 'coblocks-masonry' . COBLOCKS_ASSET_SUFFIX . '.js',
+			array( 'jquery', 'masonry', 'imagesloaded' ),
+			COBLOCKS_VERSION,
+			true
+		);
+	}
+
+	// Carousel block.
+	if ( has_block( 'coblocks/gallery-carousel', $header . $footer ) ) {
```

Comment:
Are there any other blocks that need client JS? Atomic sites will get the full set of CoBlocks, so wondering if we need to check for those.

Reply:
Just these two: https://github.com/godaddy/coblocks/blob/530d8a3df2bae78a2364c11ac06be3e263e888f5/includes/class-coblocks-block-assets.php#L142-L171

Actually, the gallery carousel block is not enabled in Simple sites, but we handle it here in order to support AT sites.